src/Security/Authentication/OpenIdConnect/src/OpenIdConnectConfigureOptions.cs

```diff
@@ -0,0 +1,157 @@
+        options.ResponseType = configSection[nameof(options.ResponseType)] ?? options.ResponseType;
+        options.ReturnUrlParameter = configSection[nameof(options.ReturnUrlParameter)] ?? options.ReturnUrlParameter;
+        options.SaveTokens = bool.TryParse(configSection[nameof(options.SaveTokens)], out var saveTokens)
+            ? saveTokens
+            : options.SaveTokens;
+        options.SignedOutCallbackPath = configSection[nameof(options.SignedOutCallbackPath)] ?? options.SignedOutCallbackPath;
+        options.SignedOutRedirectUri = configSection[nameof(options.SignedOutRedirectUri)] ?? options.SignedOutRedirectUri;
+        options.SignInScheme = configSection[nameof(options.SignInScheme)];
+        options.SignOutScheme = configSection[nameof(options.SignOutScheme)];
+        options.SkipUnrecognizedRequests = bool.TryParse(configSection[nameof(options.SkipUnrecognizedRequests)], out var skipUncrecognizedRequests)
+            ? skipUncrecognizedRequests
+            : options.SkipUnrecognizedRequests;
+        options.UsePkce = bool.TryParse(configSection[nameof(options.UsePkce)], out var usePkce)
+            ? usePkce
+            : options.UsePkce;
+        options.UseTokenLifetime = bool.TryParse(configSection[nameof(options.UseTokenLifetime)], out var useTokenLifetime)
+            ? useTokenLifetime
+            : options.UseTokenLifetime;
+    }
+
+    private static CookieBuilder? GetCookieFromConfig(IConfiguration cookieConfigSection)
+    {
+        CookieBuilder? cookieBuilder = default;
+        if (cookieConfigSection is not null && cookieConfigSection.GetChildren().Any())
+        {
+            cookieBuilder = new()
+            {
+                Domain = cookieConfigSection[nameof(CookieBuilder.Domain)],
+                HttpOnly = bool.TryParse(cookieConfigSection[nameof(CookieBuilder.HttpOnly)], out var httpOnly)
+                    ? httpOnly
+                    : false,
+                IsEssential = bool.TryParse(cookieConfigSection[nameof(CookieBuilder.IsEssential)], out var isEssential)
+                    ? isEssential
+                    : false,
+                Expiration = TimeSpan.TryParse(cookieConfigSection[nameof(CookieBuilder.Expiration)], out var expiration)
+                    ? expiration
+                    : null,
+                MaxAge = TimeSpan.TryParse(cookieConfigSection[nameof(CookieBuilder.MaxAge)], out var maxAge)
+                    ? maxAge
+                    : null,
+                Name = cookieConfigSection[nameof(CookieBuilder.Name)],
+                Path = cookieConfigSection[nameof(CookieBuilder.Path)],
+                SameSite = cookieConfigSection[nameof(CookieBuilder.SameSite)] switch
+                {
+                    "None" => SameSiteMode.None,
+                    "Strict" => SameSiteMode.Strict,
+                    "Unspecified" => SameSiteMode.Unspecified,
+                    _ => SameSiteMode.Lax
```

Comment:
When NonceCookie section is present but SameSite is not, it is being set to Lax which is different from the default one here. I'm wondering if it's intentional.
https://github.com/dotnet/aspnetcore/blob/69c7a87cd8f05da9c05c373edfc0337c9e864d9e/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectOptions.cs#L72

Reply:
Same thing goes for `Name `, `HttpOnly` and `IsEssential` as well.